pkg/utils/build/hacbs.go

```diff
@@ -0,0 +1,65 @@
+package build
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+func CheckSkipResult(taskRefName string, skiplist []string) bool {
+	for _, b := range skiplist {
+		if taskRefName == b {
+			return true
+		}
+	}
+	return false
+}
+
+func FetchTaskRunResult(pr *v1beta1.PipelineRun, pipelineTaskName string, result string) (string, error) {
+	for _, tr := range pr.Status.TaskRuns {
+		if tr.PipelineTaskName != pipelineTaskName {
+			continue
+		}
+		for _, trResult := range tr.Status.TaskRunResults {
+			if trResult.Name == result {
+				return strings.TrimSuffix(trResult.Value, "\n"), nil
+			}
+			if trResult.Name == "BASE_IMAGE_REPOSITORY" || trResult.Name == "BASE_IMAGE" {
+				return (trResult.Value), nil
+			}
+		}
+	}
+	return "", fmt.Errorf(
+		"result %q not found in TaskRuns of PipelineRun %s/%s", result, pr.ObjectMeta.Namespace, pr.ObjectMeta.Name)
+}
+
+func ValidateTaskRunResults(taskname string, result string) bool {
+	var re = regexp.MustCompile(`devfile/python`)
+	if taskname == "sanity-inspect-image" {
+		if !(re.MatchString(result)) {
+			Fail(fmt.Sprintf("Expected Result for Taskrun '%s', failed with '%s'", taskname, result))
+		}
+		return true
+	}
+	var testOutput map[string]interface{}
+	err := json.Unmarshal([]byte(result), &testOutput)
+	if err != nil {
+		Fail(fmt.Sprintf("Taskrun '%s' has failed with '%s'", taskname, err))
+	}
+	// If the test result isn't either SUCCESS or SKIPPED, the overall outcome is a failure
+	if taskname == "sast-go" || taskname == "sast-java-sec-check" {
+		if !(testOutput["result"] == "SUCCESS" || testOutput["result"] == "SKIPPED") {
+			Fail(fmt.Sprintf("Expected Result for Taskrun '%s' is SUCCESS, failed with '%s'", taskname, testOutput["failures"]))
+		}
+		return true
+	}
+	// All Other taskruns are expected to FAIL; Need to Use Valid Base Image in future
+	if testOutput["result"] == "FAILURE" {
```

Comment:
While I understand that this optimizes the flow of the code here, I think it would be easier for future updates if we separated the logic for each task by name and expressed our expectations for each of them. 
This way, if there is a change in either the base image or the task logic, we'd have a much easier time figuring out where and how to make the according change in the e2e test.

Reply:
done

Reply:
Thanks!